### ISSUE TITLE
Add property to driver definitions in manifest to allow pro only drivers included in cloud available apps

### ISSUE THREAD

**Issue by Shakesbeard** [athombv/apps-sdk-issues#224]
Rather than maintaining different apps for Homey Pro and Homey Cloud it would be nice to have an option to disable LAN specific features if the app is run in the cloud.
The possibly simplest way seems to be to disable marked drivers when the app is run in the Homey Cloud.
So I would propose adding a property to the driver entry in the manifest like
"proOnly": true

**Comment by WeeJeWel** [comment]
The 'platforms' property of a Driver is specifically made for this purpose. :)

**Comment by Shakesbeard** [comment]
Well yes, but that property, as far as I understood, defines this for the complete app as a whole.
It would be nice to have the ability to include a driver which would work in the cloud and optionally have additional drivers which would only run on pro. Or maybe I missed something?

**Comment by Shakesbeard** [comment]
Frankly, it would be even better if you could detect programmatically whether the app is running in the cloud or local on a pro. Basically to enable/disable LAN features of an app automatically based on where it is run.

**Comment by WeeJeWel** [comment]
> Well yes, but that property, as far as I understood, defines this for the complete app as a whole.
> 
> It would be nice to have the ability to include a driver which would work in the cloud and optionally have additional drivers which would only run on pro. Or maybe I missed something?

The platforms property can be defined on App, but also on a Driver, and even on a Flow card.

**Comment by WeeJeWel** [comment]
> Frankly, it would be even better if you could detect programmatically whether the app is running in the cloud or local on a pro. Basically to enable/disable LAN features of an app automatically based on where it is run.

We need this information beforehand to know if a driver can be used on someone's Homey before starting the pair sequence :)

**Comment by Shakesbeard** [comment]
> > Well yes, but that property, as far as I understood, defines this for the complete app as a whole.
> > It would be nice to have the ability to include a driver which would work in the cloud and optionally have additional drivers which would only run on pro. Or maybe I missed something?
> 
> The platforms property can be defined on App, but also on a Driver, and even on a Flow card.

Oh. That nice. I guess that literally is what I wanted to ask for :D
